hibernate-core/src/main/java/org/hibernate/type/format/jackson/JacksonXmlFormatMapper.java

```diff
@@ -24,10 +32,13 @@ public final class JacksonXmlFormatMapper implements FormatMapper {
 	private final ObjectMapper objectMapper;
 
 	public JacksonXmlFormatMapper() {
-		this(new XmlMapper());
+		this( new XmlMapper() );
 	}
 
 	public JacksonXmlFormatMapper(ObjectMapper objectMapper) {
+		// needed to automatically find and register Jackson's jsr310 module for java.time support
+		objectMapper.findAndRegisterModules();
+		objectMapper.configure( SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false );
```

Comment:
We should only do that for the `XmlMapper` that we create ourselves.

Reply:
Just as a note, a mapper we receive would need to be configured like we expect to handle this cases (i.e. string / datetime arrays).

Reply:
Yeah, I don't expect people to simply extend this class and passing something without first looking into what the base class does, so this is fine. Might be worth making `createXmlMapper` `protected` though along with the rebase as soon as https://github.com/hibernate/hibernate-orm/pull/6268 is merged

Reply:
The class is declared as `final` though, so it seems we aren't expecting anyone to extend this.